File: app/src/main/java/com/mcal/disassembler/nativeapi/Dumper.java
package com.mcal.disassembler.nativeapi;

import java.util.Vector;

public class Dumper {
    public static Vector<DisassemblerSymbol> symbols = new Vector<>();
    public static Vector<DisassemblerVtable> exploed = new Vector<>();
    public static Vector<DisassemblerClass> classes = new Vector<>();

    public static void readData(DumperListener listener) {
        symbols.clear();
        exploed.clear();
        classes.clear();
        final long size = DisassemblerDumper.getSize();
        for (int i = 0; i < size; ++i) {
            listener.updateProgress(i, (int) size);
            String demangledName = DisassemblerDumper.getDemangledNameAt(i);
            if (demangledName == null || demangledName.isEmpty() || demangledName.equals(" ")) {
                demangledName = DisassemblerDumper.getNameAt(i);
            }
            DisassemblerSymbol newSymbol = new DisassemblerSymbol(DisassemblerDumper.getNameAt(i), demangledName, DisassemblerDumper.getTypeAt(i), DisassemblerDumper.getBindAt(i));
            symbols.addElement(newSymbol);
        }
    }

    public interface DumperListener {
        void updateProgress(int last, int total);
    }
}

File: app/src/main/java/com/mcal/disassembler/nativeapi/Dumper.kt
package com.mcal.disassembler.nativeapi

import com.mcal.disassembler.interfaces.DialogProgressListener
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.withContext
import java.util.Vector

object Dumper {
    @JvmField
    var symbols = Vector<DisassemblerSymbol>()

    @JvmField
    var exploed = Vector<DisassemblerVtable>()

    @JvmField
    var classes = Vector<DisassemblerClass>()
    suspend fun readData(listener: DialogProgressListener) {
        symbols.clear()
        exploed.clear()
        classes.clear()
        val size = DisassemblerDumper.getSize()
        for (i in 0 until size) {
            withContext(Dispatchers.Main) {
                listener.updateDialogProgress(i.toInt(), size.toInt())
            }
            var demangledName = DisassemblerDumper.getDemangledNameAt(i)
            if (demangledName.isNullOrEmpty() || demangledName == " ") {
                demangledName = DisassemblerDumper.getNameAt(i)
            }
            val newSymbol = DisassemblerSymbol(
                DisassemblerDumper.getNameAt(i), demangledName, DisassemblerDumper.getTypeAt(
                    i
                ), DisassemblerDumper.getBindAt(i)
            )
            symbols.addElement(newSymbol)
        }
    }
}
